### PR TITLE
Bug 1703595: oauth/user/authz: tighten API validation checks

### PR DESCRIPTION
**Pull request by stlaz**
This PR makes changes to the validation of the authn-owned objects and tightens the edges around them.

cc @openshift/sig-auth @enj

### CODE REVIEW

**Comment by stlaz** [comment]
/retest
aws-infra

**Comment by eparis** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1703595 right?

File: test/integration/groups_test.go [hunk 1]
```diff
@@ -60,23 +60,14 @@ func TestBasicUserBasedGroupManipulation(t *testing.T) {
...
-	_, err = clusterAdminUserClient.Users().Update(firstValerie)
+	theGroup := &userapi.Group{}
+	theGroup.Name = "theGroup"
+	theGroup.Users = append(theGroup.Users, "valerie")
+	_, err = clusterAdminUserClient.Groups().Create(theGroup)
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	// make sure that user/~ returns system groups for backed users when it merges
```
Comment (enj) [inline test/integration/groups_test.go]: We still want this check, just look for `[]string{"system:authenticated", "system:authenticated:oauth"}` only.

File: test/integration/oauth_serviceaccount_client_test.go [hunk 2]
```diff
@@ -160,26 +160,8 @@ func TestOAuthServiceAccountClient(t *testing.T) {
 			t.Fatalf("Unexpected error: %v", err)
 		} else if !reflect.DeepEqual(clientAuth.Scopes, []string{"user:full"}) {
 			t.Fatalf("Unexpected scopes: %v", clientAuth.Scopes)
-		} else {
-			// update the authorization to not contain any approved scopes
-			clientAuth.Scopes = nil
```
Comment (enj) [inline test/integration/oauth_serviceaccount_client_test.go]: Set this to something minimal like `user:info` and leave the rest of the test intact.

**Comment by enj** [comment]
> https://bugzilla.redhat.com/show_bug.cgi?id=1703595 right?

Updated PR title to include it.

**Comment by enj** [comment]
> Two minor changes.

@stlaz I am OK with these as follow-ups.  @mrogers950 please LGTM unless you see something critically wrong.

**Pull request merged**